packages/expo-ads-facebook/ios/EXAdsFacebook/EXAdSettingsManager.m

```diff
@@ -111,7 +112,9 @@ - (void)onAppBackgrounded
 {
   [FBAdSettings setIsChildDirected:NO];
   [FBAdSettings setMediationService:@""];
-  [FBAdSettings setUrlPrefix:@""];
+  if (_urlPrefix) {
+    [FBAdSettings setUrlPrefix:nil];
```

Comment:
Is there any downside to always setting the URL prefix to nil? I am thinking of a case where someone changes this code in a way where `urlPrefix` starts non-nil, then is set to nil (this write is bad), and then the app is backgrounded and we don't call `setUrlPrefix: nil`.

Reply:
I don't think there is… It's:
- providing clean state for another experience (I am still torn if this should be moved to `EXScopedAdSettingsManager` or not, but it [always worked like this](https://github.com/expo/expo/blob/4828595a17548fcbb626e8c65d8988c25b1e2be1/packages/expo-ads-facebook/android/src/main/java/expo/modules/ads/facebook/AdSettingsManager.java#L109-L114) on Android without any problems, so 🤷‍♂)
- it doesn't trigger the call to Facebook backend

And regarding your scenario if someone changes the code in a way where `urlPrefix` starts non-nil… it will be their mistake and I don't know if we can prepare ourselves this kind of bug that could happen in the future right now. Do you have something on your mind? 🤔

Reply:
What I mean is: what if we removed the `if (_urlPrefix)` check and just always set it to nil? That way we'd ensure we always clear it.

Reply:
Every `setUrlPrefix` call triggers a call to Facebook, I doubt it we want to trigger a call every time an app backgrounds… I tried to shed some more light on why the calls are implemented like this in new comments, let me know if they make the matter clearer. 🙂